test/test_estimators/test_estimators.py

```diff
@@ -496,6 +564,151 @@ def exclude(lst, s):
                 assert all(leaderboard["ensemble_weight"] > 0)
 
 
+@pytest.mark.parametrize(
+    "estimator_type,dataset_name,metrics",
+    [
+        (
+            AutoSklearnClassifier,
+            "iris",
+            (autosklearn.metrics.accuracy, autosklearn.metrics.balanced_accuracy),
+        ),
+        (
+            AutoSklearnRegressor,
+            "boston",
+            (autosklearn.metrics.r2, autosklearn.metrics.root_mean_squared_error),
+        ),
+    ],
+)
+def test_leaderboard_multi_objective(
+    tmp_dir: str,
+    estimator_type: Type[AutoSklearnEstimator],
+    dataset_name: str,
+    metrics: Sequence[autosklearn.metrics.Scorer],
+):
+    # Comprehensive test tasks a substantial amount of time, manually set if
+    # required.
+    MAX_COMBO_SIZE_FOR_INCLUDE_PARAM = 3  # [0, len(valid_columns) + 1]
+    column_types = AutoSklearnEstimator._leaderboard_columns(num_metrics=2)
+
+    # Create a dict of all possible param values for each param
+    # with some invalid one's of the incorrect type
+    include_combinations = itertools.chain(
+        itertools.combinations(column_types["all"], item_count)
+        for item_count in range(1, MAX_COMBO_SIZE_FOR_INCLUDE_PARAM)
+    )
+    valid_params = {
+        "detailed": [True, False],
+        "ensemble_only": [True, False],
+        "top_k": [-10, 0, 1, 10, "all"],
+        "sort_by": [
+            "cost",
+            "cost_0",
+            "cost_1",
+            ["cost_1", "cost_0"],
+            *column_types["all"],
+            "invalid",
+        ],
+        "sort_order": ["ascending", "descending", "auto", "invalid", None],
+        "include": itertools.chain([None, "invalid", "type"], include_combinations),
+    }
+
+    # Create a generator of all possible combinations of valid_params
+    params_generator = iter(
+        dict(zip(valid_params.keys(), param_values))
+        for param_values in itertools.product(*valid_params.values())
+    )
+
+    X_train, Y_train, _, _ = putil.get_dataset(dataset_name)
+    model = estimator_type(
+        time_left_for_this_task=30,
+        per_run_time_limit=5,
+        tmp_folder=os.path.join(tmp_dir, "backend"),
+        seed=1,
+        metric=metrics,
+    )
+
+    model.fit(X_train, Y_train)
+
+    for params in params_generator:
+        # Convert from iterator to solid list
+        if params["include"] is not None and not isinstance(params["include"], str):
+            params["include"] = list(params["include"])
+
+        # Invalid top_k should raise an error, is a positive int or 'all'
+        if not (params["top_k"] == "all" or params["top_k"] > 0):
+            with pytest.raises(ValueError):
+                model.leaderboard(**params)
+
+        # Invalid sort_by column
+        elif (
+            params["sort_by"] not in column_types["all"]
+            and params["sort_by"] != "cost"
+            and params["sort_by"] != ["cost_1", "cost_0"]
```

Comment:
[Blocking]
I guess this should also check `and params["sort_by"] not in ["cost_1", "cost_0"]` since they are both individually in the `valid_params`.

Reply:
Added this check.